### PR TITLE
Drop Kubernetes v1.25, and Support Kubernetes v1.28

### PR DESCRIPTION
**Pull request by tenzen-y**
**What this PR does / why we need it**:
I updated supporting Kubernetes versions.

**Which issue(s) this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:
Part-of #2256

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

File: cmd/katib-controller/v1beta1/main.go [hunk 1]
```diff
@@ -110,7 +111,9 @@ func main() {
 
 	// Create a new katib controller to provide shared dependencies and start components
 	mgr, err := manager.New(cfg, manager.Options{
-		MetricsBindAddress:     initConfig.ControllerConfig.MetricsAddr,
+		Metrics: metricsserver.Options{
+			BindAddress: initConfig.ControllerConfig.MetricsAddr,
+		},
```
Comment (tenzen-y) [inline cmd/katib-controller/v1beta1/main.go:116]: I addressed this breaking: https://github.com/kubernetes-sigs/controller-runtime/pull/2407

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -38,6 +38,7 @@ echo ">> Using ${CODEGEN_PKG} for the code generator"
 
 # Ensure we can execute.
 chmod +x "${CODEGEN_PKG}/generate-groups.sh"
+chmod +x "${CODEGEN_PKG}/generate-internal-groups.sh"
```
Comment (tenzen-y) [inline hack/update-codegen.sh:41]: I addressed this error:

```shell
/Users/tenzen/go/pkg/mod/k8s.io/code-generator@v0.28.8/generate-groups.sh: line 66: /Users/tenzen/go/pkg/mod/k8s.io/code-generator@v0.28.8/generate-internal-groups.sh: Permission denied
pkg/apis/apis.go:18: running "../../hack/update-codegen.sh": exit status 126
make: *** [generate] Error 1
```

**Review by andreyvelich — APPROVED** [review]
Thank you for updating the Kubernetes version @tenzen-y!
I noticed that we should also update the website: https://www.kubeflow.org/docs/components/katib/hyperparameter/#prerequisites
Version is very out-of-date there.
/lgtm
/approve

**Comment by tenzen-y** [comment]
> Thank you for updating the Kubernetes version @tenzen-y! I noticed that we should also update the website: https://www.kubeflow.org/docs/components/katib/hyperparameter/#prerequisites Version is very out-of-date there. /lgtm /approve

Oh, I see. I will submit a PR to the website.

**Pull request merged**